src/freenet/node/useralerts/UserAlertManager.java

```diff
@@ -223,11 +225,35 @@ public HTMLNode renderAlert(UserAlert userAlert) {
 		userAlertNode.addChild("div", "class", "infobox-header", userAlert.getTitle());
 		HTMLNode alertContentNode = userAlertNode.addChild("div", "class", "infobox-content");
 		alertContentNode.addChild(userAlert.getHTMLText());
+		if (userAlert.canReply()) {
+			alertContentNode.addChild(renderReplyButton(userAlert, userAlert.getSourceNode()));
+		}
 		alertContentNode.addChild(renderDismissButton(userAlert, null));
 
 		return userAlertNode;
 	}
 
+	public HTMLNode renderReplyButton(UserAlert userAlert, WeakReference<PeerNode> peerNode) {
+		HTMLNode form = new HTMLNode("form",
+					new String[]{"method", "action"},
+					new String[]{"post", "/send_n2ntm/"});
+			form.addChild("input",
+					new String[]{"hidden", "name", "value"},
+					new String[]{"true", "replyTo", userAlert.getMessageText()
+							.replaceAll("\n", "NEWLINE")});
+			form.addChild("input",
+					new String[]{"hidden", "name", "value"},
+					new String[]{"true", "peernode_hashcode", Integer.valueOf(peerNode.get().hashCode()).toString()});
```

Comment:
`peerNode.get()` would be `null` if the peer node has been garbage collected, this would throw a NPE .

Reply:
fixed -- thank you!